doc/user/cluster_tls.md

```diff
@@ -0,0 +1,73 @@
+# Cluster TLS Guide
+
+Cluster TLS policy is configured on a per-cluster basis via the TPR spec provided to etcd-operator.
+
+```yaml
+apiVersion: "etcd.coreos.com/v1beta1"
+kind: "Cluster"
+metadata:
+  name: "example-etcd-cluster"
+spec:
+  ...
+  clusterTLS:
+    ....
+```
+
+For a review etcd's TLS support and requirements, please read the relevant section from the [operations guide](https://coreos.com/etcd/docs/latest/op-guide/security.html).
+
+
+## Static Cluster TLS Policy
+
+```yaml
+apiVersion: "etcd.coreos.com/v1beta1"
+kind: "Cluster"
+metadata:
+  name: "example-etcd-cluster"
+ spec:
+  ...
+  clusterTLS:
+    static:
+      serverSecretName: server-tls-secret
+      clientSecretName: client-tls-secret
+
+```
+
+* **serverSecretName**: contains pem-encoded private keys and x509 certificates needed by the etcd server.
+
+  etcd-operator will mount this secret at `/etc/etcd-operator/server-tls` for each etcd member pod in the cluster.
+
+  The server TLS assets are expected to conform to the following structure:
+
+  ```text
+  /etc/etcd-operator/server-tls/
+
+       server-client-crt.pem
```

Comment:
> Also what are the expected SANS for these certificates, or does it not matter?

On this topic, I really recommend not using IP SANS as we are now. It will really make assigning x509 identities to cluster pods a real pain.

  * Neither pod or service IP addresses are really reservable in Kubernetes. Which means you can't create the certificate until after the pod/service is created, which means you can't really pre-sign certificates in any way

  * The certificate will not be transferable if that etcd node finds itself at a new IP

  * There's no way to sign a certificate for an open-ended group of addresses. So you have to account for all possible hostnames when you sign the cert, which again is problematic. Using DNS you can sign a cert for `*.xxxx.svc.cluster.local`, which is in plain english "any service address in the xxxx kubernetes namespace". This is *really critical* stuff to be able to express to `etcd-operator` from the get-go.

To get past these problems, we need to make sure `etcd-operator`, `backup` and all the etcd member pods *only* address each other via fully-qualified cluster DNS name. For example:

`https://etcd-cluster-0002.my-namespace.svc.cluster.local`

This provides some much-needed flexibility to the end user, such as the ability to sign a generic certificate for all etcd member nodes in a reasonable way. Using IP SANS, this wouldn't really be possible as we would have to know the member IP addresses ahead of time.

A certificate could be generated for `*.my-namespace.svc.cluster.local`, and that certificate will only validate for things running in `my-namespace` on a Kubernetes cluster. I feel this is how a lot of external PKI will initially need to interact with managed components running on clusters, especially if we're not getting unique certificates signed on demand for new member pods.